x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/three_way_diff/final_readonly/storybook/mocks.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { DataView } from '@kbn/data-views-plugin/common';
+import { DataSourceType, KqlQueryType } from '../../../../../../../../common/api/detection_engine';
+import type {
+  DiffableAllFields,
+  SavedKqlQuery,
+} from '../../../../../../../../common/api/detection_engine';
+
+export const filters = [
+  {
+    meta: {
+      disabled: false,
+      negate: false,
+      alias: null,
+      index: 'logs-*',
+      key: '@timestamp',
+      field: '@timestamp',
+      value: 'exists',
+      type: 'exists',
+    },
+    query: {
+      exists: {
+        field: '@timestamp',
+      },
+    },
+    $state: {
+      store: 'appState',
+    },
+  },
+];
+
+export const savedQueryResponse = {
+  id: 'fake-saved-query-id',
+  attributes: {
+    title: 'Fake Saved Query',
+    description: '',
+    query: {
+      query: 'file.path: "/etc/passwd" and event.action: "modification"',
+      language: 'kuery',
+    },
+    filters,
+  },
+  namespaces: ['default'],
+};
+
+export const inlineKqlQuery: DiffableAllFields['kql_query'] = {
+  type: KqlQueryType.inline_query,
+  query: 'event.action: "user_login" and source.ip: "192.168.1.100"',
+  language: 'kuery',
+  filters,
+};
+
+export const savedKqlQuery: SavedKqlQuery = {
+  type: KqlQueryType.saved_query,
+  saved_query_id: 'fake-saved-query-id',
+};
+
+export const eqlQuery: DiffableAllFields['eql_query'] = {
+  query: 'process where process.name == "powershell.exe" and process.args : "* -EncodedCommand *"',
+  language: 'eql',
+  filters,
+};
+
+export const dataSourceWithIndexPatterns: DiffableAllFields['data_source'] = {
+  type: DataSourceType.index_patterns,
+  index_patterns: ['logs-*'],
+};
+
+export const dataSourceWithDataView: DiffableAllFields['data_source'] = {
+  type: DataSourceType.data_view,
+  data_view_id: 'logs-*',
+};
+
+type DataViewDeps = ConstructorParameters<typeof DataView>[0];
+
+export function mockDataView(spec: Partial<DataViewDeps['spec']> = {}): DataView {
+  const dataView = new DataView({
+    spec: {
+      title: 'logs-*',
+      fields: {
+        '@timestamp': {
+          name: '@timestamp',
+          type: 'date',
+        },
+      },
+      ...spec,
+    },
+    fieldFormats: {
+      getDefaultInstance: () => ({
+        toJSON: () => ({}),
+      }),
+    },
+  } as unknown as DataViewDeps);
```

Comment:
nit: Casting surface could be reduced by adding `searchable: true` and `aggregatable: true` to `@timestamp` field declaration. Only `fieldFormats` would need such casting.

Reply:
Thanks! Updated in this [commit](https://github.com/nikitaindik/kibana/commit/e50ece0b077172eee47edec6eb5e51959da21c93).